src/solver/sideWPointSelect.ml

```diff
@@ -0,0 +1,185 @@
+(** Strategies for widening leaf unknowns *)
+
+open Batteries
+open ConstrSys
+open Messages
+
+module type S =
+  functor (S:EqConstrSys) ->
+  functor (HM:Hashtbl.S with type key = S.v) ->
+  functor (VS:Set.S with type elt = S.v) ->
+  sig
+    type data
+
+    (** Create data required by this widening point selection strategy.
+        The parameters are not necessarily used by all strategies. 
+        @param is_stable This callback should return whether an unknown is stable.
+        @param add_infl  Allows the strategy to record additional influences.
+                          This is mainly intended for strategies like unstable-self,
+                          which records the influence of a side-effecting unknown x to the leaf y.
+    *)
+    val create_data: (S.v -> bool) -> (S.v -> S.v -> unit) -> data
+    (** Notifies this strategy that a side-effect has occured.
+        This allows the strategy to adapt its internal data structure.
+        @param data The internal state of this strategy
+        @param x    The optional source of the side-effect
+        @param y    The leaf receiving the side-effect
+    *)
+    val notify_side: data -> S.v option -> S.v -> unit
```

Comment:
There should be empty lines to separate unrelated declarations and documentation. Currently it's ambiguous which documentation belongs to which function (and odoc will complain).